src/pykeen/models/unimodal/rgcn/model.py

```diff
@@ -0,0 +1,302 @@
+# -*- coding: utf-8 -*-
+
+"""Implementation of the R-GCN model."""
+
+import logging
+from os import path
+from typing import Any, Mapping, Optional
+
+import torch
+from class_resolver import Hint, Resolver
+from torch import nn
+
+from .decompositions import Decomposition, decomposition_resolver
+from .weightings import EdgeWeighting, edge_weight_resolver
+from ...nbase import ERModel, EmbeddingSpecificationHint
+from ....nn import EmbeddingSpecification, RepresentationModule
+from ....nn.modules import Interaction, interaction_resolver
+from ....triples import TriplesFactory
+from ....typing import Initializer, RelationRepresentation
+
+__all__ = [
+    "RGCN",
+]
+
+logger = logging.getLogger(name=path.basename(__file__))
+
+
+class Bias(nn.Module):
+    """A module wrapper for adding a bias."""
+
+    def __init__(self, dim: int):
+        """Initialize the module.
+
+        :param dim: >0
+            The dimension of the input.
+        """
+        super().__init__()
+        self.bias = nn.Parameter(torch.empty(dim), requires_grad=True)
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        """Reset the layer"s parameters."""
+        nn.init.zeros_(self.bias)
+
+    # pylint: disable=arguments-differ
+    def forward(self, x: torch.FloatTensor) -> torch.FloatTensor:
+        """Add the learned bias to the input.
+
+        :param x: shape: (n, d)
+            The input.
+
+        :return:
+            x + b[None, :]
+        """
+        return x + self.bias.unsqueeze(dim=0)
+
+
+# TODO: Move to utils
```

Comment:
Maybe we should move this resolver to utils? We could also restrict the classes to a specific set of activation functions.

Reply:
Yes. Do it.

Reply:
done in 6a36eb844d6abde6b5f44e0967ee5615acf0c8dd